tests/test_tally.py

```diff
@@ -0,0 +1,213 @@
+"""Continuous integration tests with OpenMC."""
+
+import logging
+from abc import ABCMeta, abstractmethod
+from functools import cached_property
+from pathlib import Path
+from typing import Sequence, Type
+
+import build123d as bd
+import numpy as np
+import openmc
+import openmc.stats
+import pytest
+import stellarmesh as sm
+
+RELATIVE_TOLERANCE_PERCENT = 2.5 / 100
+
+logger = logging.getLogger(__name__)
+
+
+class Model(metaclass=ABCMeta):
+    @cached_property
+    @abstractmethod
+    def materials(self) -> openmc.Materials:
+        """Generate the neutron source."""
+        ...
+
+    @abstractmethod
+    def tallies(self) -> openmc.Tallies:
+        """Generate the tallies."""
+        ...
+
+    @abstractmethod
+    def source(self) -> openmc.IndependentSource:
+        """Generate the neutron source."""
+        ...
+
+    @abstractmethod
+    def csg(self) -> openmc.Model:
+        """Generate the CSG model."""
+        ...
+
+    @abstractmethod
+    def dagmc(self) -> sm.DAGMCModel:
+        """Generate the DAGMC model."""
+        ...
+
+    def _make_dagmc_model(
+        self,
+        solids: Sequence[bd.Solid],
+        materials: openmc.Materials,
+        min_mesh_size: float,
+        max_mesh_size,
+    ) -> sm.DAGMCModel:
+        mat_names: list[str] = [m.name for m in self.materials]
+        geom = sm.Geometry(solids, mat_names)
+        msh = sm.Mesh.from_geometry(geom, min_mesh_size, max_mesh_size)
+        return sm.DAGMCModel.from_mesh(msh)
+
+
+class NestedSpheres(Model):
+    """Two nested spheres."""
+
+    radius1 = 20
+    radius2 = 5
+
+    @cached_property
+    def materials(self):
+        mat1 = openmc.Material(name="1")
+        mat1.add_nuclide("Fe56", 1)
+        mat1.set_density("g/cm3", 1)
+
+        mat2 = openmc.Material(name="2")
+        mat2.add_nuclide("Be9", 1)
+        mat2.set_density("g/cm3", 1)
+        return openmc.Materials([mat1, mat2])
+
+    def tallies(self):
+        mat1_filter = openmc.MaterialFilter(self.materials[0])
+        tally1 = openmc.Tally(name="mat1_flux_tally")
+        tally1.filters = [mat1_filter]
+        tally1.scores = ["flux"]
+
+        mat2_filter = openmc.MaterialFilter(self.materials[1])
+        tally2 = openmc.Tally(name="mat2_flux_tally")
+        tally2.filters = [mat2_filter]
+        tally2.scores = ["flux"]
```

Comment:
These could be combined by having a single `MaterialFilter` that specifies both materials, but I see that maybe you just want a single value for each